### PR TITLE
Implemented Dark Iron Ale and the Morja event

### PR DESCRIPTION
**Pull request by GuybrushGit**
Issue: https://github.com/LightsHope/issues/issues/5

Added an EVENT_T_MOVEMENT_INFORM event to prevent having to hardcode the Patrons.
Most of it is done but need help with a couple of things:

- Need a condition to change Morja's gossip when the event is active, is there a way make an NPC_FLAG condition? Or a nearby NPC condition?

- The hardcoded pets could be moved to the DB, all we need is a command to remove a nearby object. Is this possible with the buddy system somehow?

### CODE REVIEW

**Comment by ratkosrb** [comment]
By change her gossip, do you mean change the textid used, or add/remove NPC_FLAG_GOSSIP?

In `gossip_menu` you can have multiple entries with the same entry but different textId and condition.

menus

If you want to add/remove gossip npc flag, that can be done with SCRIPT_COMMAND_MODIFY_FLAGS.

npc flags

Also in `gossip_menu_option` you can set required npc flag for specific option in `npc_option_npcflag` column.

For despawning a gameobject, there is no generic command currently but i can add one. There is SCRIPT_COMMAND_OPEN_DOOR, SCRIPT_COMMAND_CLOSE_DOOR , SCRIPT_COMMAND_ACTIVATE_OBJECT.

Are you sure the creature is not supposed to use the Mug object, which makes it despawn?
https://github.com/ratkosrb/server/blob/development/src/game/Maps/ScriptCommands.cpp#L566

I will add a condition for checking for nearby creature/object. You can find the conditions here now, i moved them to a separate file.
https://github.com/ratkosrb/server/blob/development/src/game/Conditions.h
https://github.com/ratkosrb/server/blob/development/src/game/Conditions.cpp

**Comment by GuybrushGit** [comment]
I'm just not sure which type of condition would be more useful in the future, one that checked an npc is nearby or one that checked if the source has certain unit flags turned on. Both can acomplish what I want.  

The object is a trap dont know if activating it will despawn it, have to test it later.

**Comment by ratkosrb** [comment]
I added the conditions you needed. I think i am going to add a flags column to conditions, so we can swap targets with it, instead of having a separate conditions for checking source and target.

File: src/game/Maps/ScriptCommands.cpp [hunk 1]
```diff
@@ -1195,5 +1195,28 @@ bool Map::ScriptCommand_RemoveItem(ScriptAction& step, Object* source, Object* t
...
+    return false;
+}
+
+// SCRIPT_COMMAND_REMOVE_OBJECT (41)
+bool Map::ScriptCommand_RemoveGameObject(ScriptAction& step, Object* source, Object* target)
+{
+    Unit* pUser = nullptr;
+    GameObject *pGo = nullptr;
+
+    if (!((pUser = ToUnit(source)) || (pUser = ToUnit(target))))
```
Comment (ratkosrb) [inline src/game/Maps/ScriptCommands.cpp]: What is the point of pUser when you are not using it. Looks like a copy-paste mistake?

**Comment by GuybrushGit** [comment]
Activate object was not usable in this case, created SCRIPT_COMMAND_REMOVE_OBJECT.

Morja and the Patrons are now all in the DB but had to leave the minipets hardcoded, giving them an EventAI would take over their PetAI and they wouldn't follow the owner around.

File: src/game/Maps/ScriptCommands.cpp (hunk 1 above)
Comment (GuybrushGit) [reply src/game/Maps/ScriptCommands.cpp]: Yeah fixed it.

**Comment by GuybrushGit** [comment]
Welp... just realized AI script IDs are creature entry + XX

**Comment by ratkosrb** [comment]
Yep, its not just random numbers.

**Pull request merged**